pkg/daemon/update.go

```diff
@@ -2155,24 +2155,20 @@ func (dn *Daemon) reboot(rationale string) error {
 	}
 	dn.logSystem("initiating reboot: %s", rationale)
 
-	rebootCmd := rebootCommand(rationale)
-
 	// reboot, executed async via systemd-run so that the reboot command is executed
 	// in the context of the host asynchronously from us
 	// We're not returning the error from the reboot command as it can be terminated by
 	// the system itself with signal: terminated. We can't catch the subprocess termination signal
 	// either, we just have one for the MCD itself.
+	rebootCmd := rebootCommand(rationale)
 	if err := rebootCmd.Run(); err != nil {
 		dn.logSystem("failed to run reboot: %v", err)
 		mcdRebootErr.Inc()
+		return fmt.Errorf("reboot command failed, something is seriously wrong")
 	}
-
-	// wait to be killed via SIGTERM from the kubelet shutting down
-	time.Sleep(defaultRebootTimeout)
-
-	// if everything went well, this should be unreachable.
-	mcdRebootErr.Inc()
-	return fmt.Errorf("reboot failed; this error should be unreachable, something is seriously wrong")
+	// if we're here, reboot went through successfully
+	dn.logSystem("reboot successful")
```

Comment:
So in theory, it is possible for the MCO to run another sync after this returns? i.e. If the reboot is slow enough to sigterm'ing the MCO, is it possible for us to have started another update?

Looking at our other use the the timeout we do have in `RunFirstbootCompleteMachineconfig`, looks like that doesn't do quite the same thing, but we use the timeout to stop the MCD from doing another sync. We could do that for extra extra safety if we wanted to 

I'm fine with us not worrying about that path in this PR as well, just a thought

Reply:
I too have some doubt here on whether to keep some sleep duration. But, I am not convinced (or have enough data) on what purpose it will serve by adding some sleep before reboot. Also, if we want to keep some sleep duration, on what parameter we determine how much we should keep.

@cgwalters Perhaps, you may have some more insight on does it makes sense to keep `time.Sleep(defaultRebootTimeout)` considering we no longer explicitly stop kubelet service with RR https://github.com/openshift/machine-config-operator/pull/2509

Reply:
Agree we can remove the sleep here; I think that was only necessary before graceful node shutdown.

Reply:
sounds good, thanks for the feedback! (: